x-pack/plugins/observability_solution/entity_manager/server/routes/enablement/disable.ts

```diff
@@ -58,10 +50,14 @@ export function disableEntityDiscoveryRoute<T extends RequestHandlerContext>({
           deleteData: req.query.deleteData,
         });
 
-        await deleteEntityDiscoveryAPIKey((await context.core).savedObjects.client);
-        await server.security.authc.apiKeys.invalidateAsInternalUser({
-          ids: [apiKey.id],
-        });
+        server.logger.debug('reading entity discovery API key from saved object');
+        const apiKey = await readEntityDiscoveryAPIKey(server);
+        if (apiKey) {
```

Comment:
api key could be manually deleted outside of entityManager apis

Reply:
maybe worth an inline comment?  

something to the effect of, "api key already been deleted; does not affect disablement flow, we just don't need to delete it here."